erc1155/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.4.2/contracts/token/ERC1155/IERC1155.sol
+
+#![no_std]
+#![feature(const_btree_new)]
+
+#[cfg(test)]
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::U256;
+use scale_info::TypeInfo;
+
+pub mod base;
+
+const GAS_AMOUNT: u64 = 300_000_000;
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug)]
+struct Erc1155Token {
+    name: String,
+    symbol: String,
+    description: String,
+    base_uri: String,
+    balances: BTreeMap<u128, BTreeMap<ActorId, u128>>,
+    operator_approvals: BTreeMap<ActorId, BTreeMap<ActorId, bool>>,
+}
+
+static mut ERC1155_TOKEN: Erc1155Token = Erc1155Token {
+    name: String::new(),
+    symbol: String::new(),
+    base_uri: String::new(),
+    description: String::new(),
+    balances: BTreeMap::new(),
+    operator_approvals: BTreeMap::new(),
+};
+
+impl Erc1155Token {
+    fn get_balance(&self, account: &ActorId, id: &u128) -> u128 {
+        *self
+            .balances
+            .get(id)
+            .and_then(|m| m.get(account))
+            .unwrap_or(&0)
+    }
+
+    fn set_balance(&mut self, account: &ActorId, id: &u128, amount: u128) {
+        debug!("before mint: {:?}", self.get_balance(account, id));
+
+        let mut _balance = self
+            .balances
+            .entry(*id)
+            .or_default()
+            .insert(*account, amount);
+
+        debug!("after mint: {:?}", self.get_balance(account, id));
+    }
+
+    fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
+        self.get_balance(account, id)
```

Comment:
What is this function for if there is already a function `get_balance`?

Reply:
I rename get_balance to _get_balance, treat it as an internal function.
the balance_of is the public function.